File: app/src/main/java/example/com/archtest/api/LoadingListener.java
package example.com.archtest.api;

public interface LoadingListener {
	/**
	 * Called when there is no more data to load from network
	 */
	void onAllDataLoaded();

	/**
	 * Called when network request failed.
	 * Here you can display error message to user if needed
	 * @param message error message
	 */
	void onLoadError(String message);
}

File: app/src/main/java/example/com/archtest/api/LoadingListener.kt
package example.com.archtest.api

interface LoadingListener {
    /**
     * Called after each page loaded
     * @param pageNumber loaded page number
     */
    fun onPageLoaded(pageNumber: Int)

    /**
     * Called when there is no more data to load from network
     */
    fun onAllDataLoaded()

    /**
     * Called when network request failed.
     * Here you can display error message to user if needed
     * @param message error message
     */
    fun onLoadError(message: String)
}